modules/tinymce/src/plugins/quickbars/main/ts/insert/Toolbars.ts

```diff
@@ -12,10 +12,16 @@ const addToEditor = (editor: Editor): void => {
       predicate: (node) => {
         const sugarNode = SugarElement.fromDom(node);
         const textBlockElementsMap = editor.schema.getTextBlockElements();
-        const isRoot = (elem: SugarElement<Node>) => elem.dom === editor.getBody();
+        const isRoot = (elem) => elem.dom === editor.getBody();
+        const isFakeCaret = (elem) =>
+          Attribute.has(elem, 'data-mce-bogus') || SelectorExists.ancestor(elem, '[data-mce-bogus="all"]', (el) => isRoot(el));
```

Comment:
```suggestion
        const isRoot = (elem: SugarElement<Node>) => elem.dom === editor.getBody();
        const isFakeCaret = (elem: SugarElement<Node>) =>
          Attribute.has(elem, 'data-mce-bogus') || SelectorExists.ancestor(elem, '[data-mce-bogus="all"]', isRoot);
```
Just copying what I mentioned in DMs. Sorry for the confusion 😞